### PR TITLE
fix bundle loop caused by recent change sort algorithm updates.

### PR DESCRIPTION
**Pull request by hmlanigan**
A loop deploying a complicated bundle was recently reported in the juju MM channel.  It turned out to be in applicationsForMachineChange.  This method relied on the old changes sort to work.  

The method was meant error reporting or debugging, therefore do not call for logging, if debug is not used.

## QA steps

The following bundle will fail to create machines, it's intended for a Maas config.  However the command should complete without error.  https://pastebin.canonical.com/p/5VsyJcDGGN/ 

It has been slightly modified from the original report, to complete without Maas spaces and tags.  It does reproduce the bug reported.

```sh
juju bootstrap localhost test 
juju deploy loop-bundle.yaml
```

Also try a bundle intended to upgrade an existing charm.

### CODE REVIEW

File: core/bundle/changes/export_test.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Copyright 2021 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package bundlechanges
```
Comment (manadart) [inline core/bundle/changes/export_test.go:4]: This should go in _package_test.go_. This package is recent enough to have been set up properly with a package-local package_test.

File: cmd/juju/application/deployer/bundlehandler_test.go [hunk 2]
```diff
@@ -55,7 +57,7 @@ var _ = gc.Suite(&BundleDeployRepositorySuite{})
 func (s *BundleDeployRepositorySuite) SetUpTest(_ *gc.C) {
 	s.deployArgs = make(map[string]application.DeployArgs)
 	s.output = bytes.NewBuffer([]byte{})
-	//logger.SetLogLevel(loggo.TRACE)
+	//logger.SetLogLevel(loggo.DEBUG)
```
Comment (manadart) [inline cmd/juju/application/deployer/bundlehandler_test.go:60]: Delete?

**Review by manadart — APPROVED** [review]
Works a treat. Couple of suggestions.

**Comment by hmlanigan** [comment]
$$merge$$

**Pull request merged**